### PR TITLE
RSC-998 donut chart docs

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://issues.sonatype.org/browse/RSC-998

Update the description tile inside of the gallery to provide information on the default behaviours of CSS height and display properties, and the options to override the defaults.

### CODE REVIEW

**Comment by EmilyBermudez88** [comment]
The first two commits (dacbc99 and 8103cd1) were from the previous ticket RSC-1084) and accidentally added.

**Comment by rpokorny** [comment]
> The first two commits (https://github.com/sonatype/sonatype-react-shared-components/commit/dacbc9967e5ec177033d2cee71152f0d5b921f79 and https://github.com/sonatype/sonatype-react-shared-components/commit/8103cd1510fe7d06e282396dfc95138e44e9149b) were from the previous ticket RSC-1084) and accidentally added.

Don't worry about that. No one looks at the commit list.  Do make sure the PR diff is correct though (it is here).

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx [hunk 1]
```diff
@@ -27,6 +27,18 @@ const NxBinaryDonutChartPage = () =>
       <NxP>
         <NxCode>NxBinaryDonutChart</NxCode> represents a binary donut chart.
       </NxP>
+      <NxP>
+        The <NxCode>NxBinaryDonutChart</NxCode> accepts a default height of 30px (without padding or margin), which
+        can be modified by accessing the CSS height property on the <NxCode>.nx-binary-donut-chart</NxCode> class.
```
Comment (rpokorny) [inline gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx:32]: We don't want to advocate for them restyling RSC classes.  Rather they should add a classname of their own and apply their styling overrides to that class.

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx [hunk 2]
```diff
@@ -27,6 +27,18 @@ const NxBinaryDonutChartPage = () =>
...
+      </NxP>
+      <NxP>
+        By default the CSS property of <NxCode>NxBinaryDonutChart</NxCode> is set to a value of
```
Comment (rpokorny) [inline gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx:35]: Is this supposed to say "display property"?

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx [hunk 3]
```diff
@@ -27,6 +27,18 @@ const NxBinaryDonutChartPage = () =>
...
+        inline. <NxCode>Display:inline</NxCode> will not generate any line breaks to maintain the flow of the
```
Comment (rpokorny) [inline gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx:36]: Code rules take precedence over sentence rules```suggestion
        inline. <NxCode>display:inline</NxCode> will not generate any line breaks to maintain the flow of the
```

**Comment by rpokorny** [comment]
A PR that has no changes to the code shouldn't need changes to the visual test snapshots.  Does your new snapshot pass in jenkins?

**Comment by rpokorny** [comment]
You could consider this to be overkill, but I'm wondering if a new example demonstrating block layout, custom sizing, and even custom colors would be a good idea.

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx (hunk 2 above)
Comment (EmilyBermudez88) [reply gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx:35]: Yes  it is, thanks for the catch

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx (hunk 3 above)
Comment (EmilyBermudez88) [reply gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartPage.tsx:36]: ok, I'll make the change

**Comment by EmilyBermudez88** [comment]
> A PR that has no changes to the code shouldn't need changes to the visual test snapshots. Does your new snapshot pass in jenkins?

I thought so too, but it failed two CI builds citing a diff in the "NxBinary Donut Chart Donut with large hole looks right" . Once I updated the snapshot the CI build passed in Jenkins

**Comment by EmilyBermudez88** [comment]
> You could consider this to be overkill, but I'm wondering if a new example demonstrating block layout, custom sizing, and even custom colors would be a good idea.

I'll leave the final decision up to you, but I'm happy to a customizable example to the page as well.

**Comment by rpokorny** [comment]
> I'll leave the final decision up to you, but I'm happy to a customizable example to the page as well.

Let's do it

**Comment by EmilyBermudez88** [comment]
> > The first two commits ([dacbc99](https://github.com/sonatype/sonatype-react-shared-components/commit/dacbc9967e5ec177033d2cee71152f0d5b921f79) and [8103cd1](https://github.com/sonatype/sonatype-react-shared-components/commit/8103cd1510fe7d06e282396dfc95138e44e9149b)) were from the previous ticket RSC-1084) and accidentally added.
> 
> Don't worry about that. No one looks at the commit list. Do make sure the PR diff is correct though (it is here).

Ok, thanks

**Comment by rpokorny** [comment]
> Would you like a new branch started for that, or am I ok to keep working within the RSC-998 ticket?

To me it's part of the ideal end state for this ticket, and isn't separate

**Comment by EmilyBermudez88** [comment]
> > Would you like a new branch started for that, or am I ok to keep working within the RSC-998 ticket?
> 
> To me it's part of the ideal end state for this ticket, and isn't separate

Perfect, I'll get started.

**Comment by EmilyBermudez88** [comment]
> > Would you like a new branch started for that, or am I ok to keep working within the RSC-998 ticket?
> 
> To me it's part of the ideal end state for this ticket, and isn't separate

The extra GalleryExampleTile has been added, but one question. As of right now, there is a separate SCSS file for the custom example, so the tsx file just has an extra class name attached. If that's the case, only the classname appears as a modification in the Live Example element. Would you rather it be inline styled for readers to see? Or should there be a second live example included? Or can readers discern the meaning for themselves?

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ */
+
+ .gallery-binary-donut-chart-custom {
+    display: block;
+    height: 45px;
+    .nx-binary-donut-chart__arc {
+      stroke: var(--nx-swatch-purple-70);
+    }
+  }
+
```
Comment (rpokorny) [inline gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.scss:15]: Trailing whitespace

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { NxBinaryDonutChart } from '@sonatype/react-shared-components';
+
+export default function NxBinaryDonutChartCustomExample() {
```
Comment (rpokorny) [inline gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.tsx:10]: Let's add a visual test for this example

File: gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.scss (hunk 4 above)
Comment (EmilyBermudez88) [reply gallery/src/components/NxBinaryDonutChart/NxBinaryDonutChartCustomExample.scss]: thanks for the catch

**Review by rpokorny — APPROVED** [review]
You need to commit the snapshot for your visual test but otherwise looks good.

**Comment by EmilyBermudez88** [comment]
> You need to commit the snapshot for your visual test but otherwise looks good.

updated. Thanks

**Pull request merged**